app/src/main/java/com/amaze/filemanager/ui/activities/MainActivity.java

```diff
@@ -586,6 +595,39 @@ private void initFabToSave(final ArrayList<Uri> uris) {
     floatingActionButton.setVisibility(View.VISIBLE);
   }
 
+  /*
+   * When saving from browser, we are fetching the Summary and the URL of the page and saving it in
+   * a text file and finally returning the URI so that can be saved.
+   * */
+  public Uri createTempTextFile(String data, String textFileName) {
```

Comment:
Few problems here:
1. Please don't write utility methods in MainActivity, try to move them somewhere else. Better place for this would be `FileUtil`
See `FileUtil#writeUriToStorage`
2. The original requirement was to save the title and url in text file. What this will do is, it'll save the text file, but will proceed to call `initFabToSave` which inturn will save the file again (on FAB on click). So you're essentially creating file before even clicking the FAB save button.

Reply:
Thanks Vishal. Will make the necessary changes, wasn't aware of the entire project structure.